src/generate.ts

```diff
@@ -17,10 +17,18 @@ type RequiredModelSubset = Pick<
 	'tables' | 'relationships' | 'synonyms'
 >;
 
+export type Options = {
+	noSerial?: boolean;
+	noMissingReferences?: boolean;
```

Comment:
I think `inlineWriteReferences` would be a better name for what it's doing, since it only applies to write and it's inlining them - the missing references thing is an issue you *might* get if you don't use the option, but is definitely no guarantee.
```suggestion
	inlineWriteReferences?: boolean;
```

Alternatively I guess we could be smart and auto-inline in cases where we see that we wouldn't generate a write typing for the referenced field, maybe that's the correct path and keep options to a minimum?

Reply:
@Page- I considered this, but auto-inlining for those cases would result in this being a breaking change. If we're good with that, I like the idea of auto-inlining and not requiring the option.

Reply:
I don't think auto-inlining is a breaking change, in fact it's the opposite surely? We would only inline in cases where if we don't then it would break due to trying to reference a type that doesn't exist

Reply:
The main downside I can see with it is if it's particularly complex to detect (which I'm hoping it wouldn't be, but maybe)

Reply:
@Page- What I mean by breaking is that for the same input the output will differ. In the sense that we're fixing the output to no longer be broken, I could see it as a patch.

Reply:
Yeah it's definitely a patch imo since it's fixing broken behavior, but in theory any bugfix could be a breaking change if you're relying on it breaking.. so really the difference between bugfix and breaking change is a judgement based one based on whether we consider it as going from broken -> working behavior or working -> (different) working

Reply:
Yeah, I'm happy with it being a patch then, but will split this into two PRs since these are solving two separate problems.